glotaran/project/project.py

```diff
@@ -0,0 +1,446 @@
+"""The glotaran project module."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from os import getcwd
+from pathlib import Path
+from typing import Any
+from typing import Literal
+
+import xarray as xr
+
+from glotaran.builtin.io.yml.utils import load_dict
+from glotaran.model import Model
+from glotaran.parameter import ParameterGroup
+from glotaran.project.project_data_registry import ProjectDataRegistry
+from glotaran.project.project_model_registry import ProjectModelRegistry
+from glotaran.project.project_parameter_registry import ProjectParameterRegistry
+from glotaran.project.project_result_registry import ProjectResultRegistry
+from glotaran.project.result import Result
+from glotaran.project.scheme import Scheme
+from glotaran.utils.ipython import MarkdownStr
+
+TEMPLATE = """version: {gta_version}
+name: {name}
+"""
+
+PROJECT_FILE_NAME = "project.gta"
+
+
+@dataclass
+class Project:
+    """A project represents a projectfolder on disk which contains a project file.
+
+    A project file is a file in `yml` format with name `project.gta`
+    """
+
+    file: Path
+    name: str
+    version: str
+
+    folder: Path
+
+    def __post_init__(self):
+        """Overwrite of post init."""
+        if isinstance(self.file, str):
+            self.file = Path(self.file)
+        if self.folder is None:
+            self.folder = self.file.parent
+        if isinstance(self.folder, str):
+            self.folder = Path(self.folder)
+
+        self._data_registry = ProjectDataRegistry(self.folder)
+        self._model_registry = ProjectModelRegistry(self.folder)
+        self._parameter_registry = ProjectParameterRegistry(self.folder)
+        self._result_registry = ProjectResultRegistry(self.folder)
+
+    @classmethod
+    def create(cls, name: str | None = None, folder: str | Path | None = None) -> Project:
```

Comment:
Same as for `open`:

> This will have a problem when using relative paths and executing from different paths.
We [already solved](https://github.com/glotaran/pyglotaran-extras/blob/f4b64beb621195d428e1bfd70c819af026889a38/pyglotaran_extras/io/setup_case_study.py#L52) this problem in the extras in a way that it works for scripts and notebooks.
Just copy that function over to `glotaran.utils.io`